drivers/media/i2c/irs1125.c

```diff
@@ -0,0 +1,1137 @@
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_CONTINUOUS_TRIG,
+        .name = "Enable/disable continuous trigger",
+        .type = V4L2_CTRL_TYPE_BOOLEAN,
+        .flags = V4L2_CTRL_FLAG_EXECUTE_ON_WRITE,
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 0
+        },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_TRIGGER,
+        .name = "Capture a single sequence",
+        .type = V4L2_CTRL_TYPE_BOOLEAN,
+        .flags = V4L2_CTRL_FLAG_EXECUTE_ON_WRITE,
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_RECONFIG,
+        .name = "Trigger imager reconfiguration",
+        .type = V4L2_CTRL_TYPE_BOOLEAN,
+        .flags = V4L2_CTRL_FLAG_EXECUTE_ON_WRITE,
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_ILLU_ON,
+        .name = "Turn illu on or off",
+        .type = V4L2_CTRL_TYPE_BOOLEAN,
+        .flags = V4L2_CTRL_FLAG_EXECUTE_ON_WRITE,
+        .min = 0,
+        .max = 1,
+        .step = 1,
+        .def = 1
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT0,
+        .name = "Get ident 0 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT1,
+        .name = "Get ident 1 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+    {
+        .ops = &irs1125_ctrl_ops,
+        .id = IRS1125_CID_IDENT2,
+        .name = "Get ident 2 information",
+        .type = V4L2_CTRL_TYPE_INTEGER,
+        .flags = V4L2_CTRL_FLAG_READ_ONLY,
+        .min = S32_MIN,
+        .max = S32_MAX,
+        .step = 1,
+        .def = 0
+    },
+};
+
+static int irs1125_detect(struct v4l2_subdev *sd)
+{
+    u16 read;
+    int ret;
+    struct i2c_client *client = v4l2_get_subdevdata(sd);
+
+    ret = irs1125_read(sd, IRS1125_REG_DESIGN_STEP, &read);
+    if (ret < 0)
+    {
+        dev_err(&client->dev, "error reading from i2c\n");
+        return ret;
+    }
+
+    if (read != 0x0a12) {
+        dev_err(&client->dev, "Design step expected 0x0a12 got %x", read);
+        return -ENODEV;
+    }
+
+    return 0;
+}
+
+static int irs1125_open(struct v4l2_subdev *sd, struct v4l2_subdev_fh *fh)
+{
+    struct v4l2_mbus_framefmt *format =
+                v4l2_subdev_get_try_format(sd, fh->pad, 0);
+    struct v4l2_rect *crop =
+                v4l2_subdev_get_try_crop(sd, fh->pad, 0);
+
+    crop->left = IRS1125_COLUMN_START_DEF;
```

Comment:
You don't do anything with crop, so remove it.

Reply:
removed